File: baremaps-config/src/main/java/com/baremaps/config/tileset/Tileset.java
/*
 * Copyright (C) 2020 The Baremaps Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */
package com.baremaps.config.tileset;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(Include.NON_NULL)
public class Tileset {

  private String tilejson = "2.2.0";

  private String name;

  private String version;

  private String description;

  private String attribution;

  private Center center;

  private Bounds bounds = new Bounds();

  private Double minZoom;

  private Double maxZoom;

  private List<String> tiles = new ArrayList<>();

  @JsonProperty("vector_layers")
  private List<Layer> layers = new ArrayList<>();

  public Tileset() {}

  public Tileset(Layer... layers) {
    this.layers = Arrays.asList(layers);
  }

  public Tileset(String name, Layer... layers) {
    this.name = name;
    this.layers = Arrays.asList(layers);
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getVersion() {
    return version;
  }

  public void setVersion(String version) {
    this.version = version;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getAttribution() {
    return attribution;
  }

  public void setAttribution(String attribution) {
    this.attribution = attribution;
  }

  public Center getCenter() {
    return center;
  }

  public void setCenter(Center center) {
    this.center = center;
  }

  public Bounds getBounds() {
    return bounds;
  }

  public void setBounds(Bounds bounds) {
    this.bounds = bounds;
  }

  public Double getMinZoom() {
    return minZoom;
  }

  public void setMinZoom(Double minZoom) {
    this.minZoom = minZoom;
  }

  public Double getMaxZoom() {
    return maxZoom;
  }

  public void setMaxZoom(Double maxZoom) {
    this.maxZoom = maxZoom;
  }

  public List<Layer> getLayers() {
    return layers;
  }

  public void setLayers(List<Layer> layers) {
    this.layers = layers;
  }

  public String getTilejson() {
    return tilejson;
  }

  public void setTilejson(String tilejson) {
    this.tilejson = tilejson;
  }

  public List<String> getTiles() {
    return tiles;
  }

  public void setTiles(List<String> tiles) {
    this.tiles = tiles;
  }
}

File: daylight/tileset.js
/**
 Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 in compliance with the License. You may obtain a copy of the License at

 http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software distributed under the License
 is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 or implied. See the License for the specific language governing permissions and limitations under
 the License.
 **/
import config from "./config.js";

import aerialway from "../basemap/layers/aerialway/tileset.js";
import aeroway from "../basemap/layers/aeroway/tileset.js";
import amenity from "../basemap/layers/amenity/tileset.js";
import attraction from "../basemap/layers/attraction/tileset.js";
import barrier from "../basemap/layers/barrier/tileset.js";
import boundary from "../basemap/layers/boundary/tileset.js";
import building from "../basemap/layers/building/tileset.js";
import highway from "../basemap/layers/highway/tileset.js";
import natural from "../basemap/layers/natural/tileset.js";
import leisure from "../basemap/layers/leisure/tileset.js";
import landuse from "../basemap/layers/landuse/tileset.js";
import railway from "../basemap/layers/railway/tileset.js";
import route from "../basemap/layers/route/tileset.js";
import man_made from "../basemap/layers/man_made/tileset.js";
import power from "../basemap/layers/power/tileset.js";
import point from "../basemap/layers/point/tileset.js";
import waterway from "../basemap/layers/waterway/tileset.js";
import coastline from "./layers/coastline/tileset.js";


export default {
  "tilejson": "2.2.0",
  "center": [...config.center, config.zoom],
  "bounds": [...config.bounds],
  "minzoom": 0.0,
  "maxzoom": 14.0,
  "tiles": [
    `${config.host}/tiles/{z}/{x}/{y}.mvt`
  ],
  attribution: '© <a href="https://www.openstreetmap.org/">OpenStreetMap</a> © <a href="https://www.geoboundaries.org">geoBoundaries</a>',
  database: config.database,
  "vector_layers": [
    aerialway,
    aeroway,
    amenity,
    attraction,
    barrier,
    boundary,
    building,
    highway,
    landuse,
    leisure,
    man_made,
    natural,
    coastline,
    point,
    power,
    railway,
    route,
    waterway,
  ]
}
